tower-util/src/combinator/or_else.rs

```diff
@@ -0,0 +1,58 @@
+use super::*;
+
+/// `Service` analog to the `futures::future::OrElse` combinator.
+///
+/// TODO: Currently restricted to not modifying the error type,
+/// since we need `poll_ready` and `call` to have the same type.
+/// But it doesn't make sense to `or_else` the `poll_ready` type.
```

Comment:
Elsewhere in cases like, we tend to have the `Error` type be `Box<dyn std::errror::Error + Send + Sync>`, and then place requirements on the inner services like: `T::Error: Into<Box<dyn std::errror::Error + Send + Sync>>`. This tends to be more ergonomic.